packages/migration/src/migrations/v0026_videoDuration.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function videoDuration({ pages, ...rest }) {
+  return {
+    pages: pages.map(reducePage),
+    ...rest,
+  };
+}
+
+function reducePage({ elements, ...rest }) {
+  return {
+    elements: elements.map(updateElement),
+    ...rest,
+  };
+}
+
+function updateElement(element) {
+  if (
+    element.type === 'video' &&
+    element.resource &&
+    element.resource?.lengthFormatted &&
+    !element.resource?.length
+  ) {
+    const times = element.resource.lengthFormatted.split(':');
+    const timesNumbers = times.map(Number);
+    const [minutes, seconds] = timesNumbers;
+    const length = 60 * minutes + seconds;
```

Comment:
Nit: this won't work for videos with duration >= 1 hour.

Reply:
Good catch! I have fix, including a unit test. https://github.com/google/web-stories-wp/pull/7691/commits/02a5891c1a6da6be3a881b7ee640bba662e4eed3